### PR TITLE
Add client to mockServer

### PR DESCRIPTION
**Pull request by sqren**
Adding mockServerClient to mockServer

@Tradeshift/collaboration

### CODE REVIEW

**Comment by sqren** [comment]
retest

**Comment by vlezhnin** [comment]
why not `http-mock-server`?

**Comment by sqren** [comment]
> why not http-mock-server?

https://www.npmjs.com/package/http-mock-server

File: bin/helpers.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
+const path = require('path');
+const listenerService = require('../src/server/listenerService.js');
+
+const helpers = {};
```
Comment (vlezhnin) [inline bin/helpers.js]: there is only reading from configs in function in this file. I would suggest it a more specific name instead of `helpers`

**Comment by vlezhnin** [comment]
what is bin folder? Can't we run from src?

File: bin/helpers.js (hunk 1 above)
Comment (sqren) [inline bin/helpers.js]: Yeah, I was a little lazy. Do you have any good suggestions?

Comment (vlezhnin) [inline bin/helpers.js]: something like `configContext` or `configContextInitializer` or `configInitializer`

File: bin/helpers.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+						console.log(err);
+						return;
+					}
+					onFileContent(filename, JSON.parse(content));
+				});
+			});
+	});
+};
+
+helpers.registerMockConfigRoutes = function (path) {
```
Comment (vlezhnin) [inline bin/helpers.js]: then you can rename this to simple `registerRoutes`

**Comment by sqren** [comment]
> what is bin folder? Can't we run from src?

We could but it's a convention to put your executables in bin folder. Webpack does the same:
https://github.com/webpack/webpack/blob/master/package.json#L76

File: bin/helpers.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
+const path = require('path');
+const listenerService = require('../src/server/listenerService.js');
+
+const helpers = {};
+
+helpers.getMockConfigs = function (dirname, onFileContent) {
```
Comment (vlezhnin) [inline bin/helpers.js]: if you rename the file you can simplify this name to `loadConfigs`

File: bin/helpers.js (hunk 2 above)
Comment (vlezhnin) [inline bin/helpers.js]: or `init`

File: src/client/Client.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	}
+
+	clear () {
+		return this.req({
+			uri: '/clear',
+			method: 'DELETE'
+		});
+	}
+
+	allowAll () {
```
Comment (vlezhnin) [inline src/client/Client.js]: we probably should rename this to `mockAllGetRequests` or remove this at all. It does not seem very useful

File: src/client/Client.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+		});
+	}
+
+	addRoutes (...routes) {
+		return Q.all(routes.map(options => this.addRoute(options)));
+	}
+
+	sendData (options) {
+		return this.req({
+			uri: '/listener/' + this.port + '/chunk',
```
Comment (vlezhnin) [inline src/client/Client.js]: maybe `/listener/${this.port}/chunk`

**Comment by vlezhnin** [comment]
🍏

**Pull request merged**